python/tests/test_mode_decomposition.py

```diff
@@ -83,6 +83,50 @@ def test_mode_decomposition(self):
         self.assertAlmostEqual(abs(coeffs[0, 0, 1])**2 / abs(incident_coeffs[0, 0, 0])**2,
                                -taper_flux[0] / incident_flux[0], places=4)
 
+    def test_oblique_waveguide_backward_mode(self):
+        sxy = 12.0
+        cell_size = mp.Vector3(sxy,sxy,0)
+
+        dpml = 0.6
+        pml_layers = [mp.PML(thickness=dpml)]
+
+        fcen = 1/1.55
+        rot_angle = np.radians(35.0)
+        kpoint = mp.Vector3(1,0,0).rotate(mp.Vector3(0,0,1), rot_angle) * -1.0
+        sources = [mp.EigenModeSource(src=mp.GaussianSource(fcen,fwidth=0.1),
+                                      center=mp.Vector3(0.5*sxy-3.4,0,0),
+                                      size=mp.Vector3(0,sxy,0),
+                                      direction=mp.NO_DIRECTION,
+                                      eig_kpoint=kpoint,
+                                      eig_band=1,
+                                      eig_parity=mp.ODD_Z,
+                                      eig_match_freq=True)]
+
+        geometry = [mp.Block(center=mp.Vector3(),
+                             size=mp.Vector3(mp.inf,1,mp.inf),
+                             e1 = mp.Vector3(1,0,0).rotate(mp.Vector3(0,0,1), rot_angle),
+                             e2 = mp.Vector3(0,1,0).rotate(mp.Vector3(0,0,1), rot_angle),
+                             material=mp.Medium(index=3.5))]
+
+        sim = mp.Simulation(cell_size=cell_size,
+                            resolution=40,
+                            boundary_layers=pml_layers,
+                            sources=sources,
+                            geometry=geometry)
+
+        mode = sim.add_mode_monitor(fcen, 0, 1,
+                                    mp.FluxRegion(center=mp.Vector3(-0.5*sxy+dpml,0,0),
+                                                  size=mp.Vector3(0,sxy,0)))
+
+        sim.run(until_after_sources=30)
+
+        flux = mp.get_fluxes(mode)[0]
+        coeff = sim.get_eigenmode_coefficients(mode,[1],
+                                               direction=mp.NO_DIRECTION,
+                                               kpoint_func=lambda f,n: kpoint).alpha[0,0,0]
+
+        print("oblique-waveguide-flux:, {:.6f}, {:.6f}".format(-flux, abs(coeff)**2))
+        self.assertAlmostEqual(-flux, abs(coeff)**2, places=1)
```

Comment:
`places=1` seems like a low tolerance.   Do they converge?

Reply:
Actually, since this is an *absolute* error, I have no idea whether it is a small or a large tolerance?  

What is the magnitude of `|flux|` here?

Reply:
This is the output of the print statement:
```
oblique-waveguide-flux:, 100.008731, 99.977897

```

Reply:
Fixed.

Reply:
Okay, so `places=1` is actually a relative tolerance of `1e-4`, which is pretty good!

Reply:
We could do
```
self.assertAlmostEqual(-1, abs(coeff)**2/flux, places=2)
```
and then maybe we could use a lower resolution.

Reply:
```
    self.assertAlmostEqual(-1,abs(coeff)**2/flux,places=3)
AssertionError: -1 != -0.9994960396371381 within 3 places
```